python/ray/experimental/async_plasma.py

```diff
@@ -23,40 +45,19 @@ def process_notifications(self, messages):
         """Process notifications."""
         for object_id, object_size, metadata_size in messages:
             if object_size > 0 and object_id in self._waiting_dict:
-                self._complete_future(object_id)
+                self._loop.call_soon_threadsafe(_complete_future, self,
```

Comment:
I'm unsure if this should be a direct `call_soon_threadsafe`, or if there should be a `call_later` with a delay. The latter option would ensure there is some delay between the initial receiving of the callback and the python `_complete_future` checking to see if the object is present.

Reply:
I’m more troubled by the two call_soon_threadsafe here, I guess because we are interacting with python future, we really need two threadsafe?

Reply:
Regarding the delay, it’s hard to quantify any artificial delay to be added, I would just go for the call soon

Reply:
I can remove the second wrapped `call_soon_threadsafe` and call the  `set_result` directly.